04-data-tree/index.js

```diff
@@ -0,0 +1,40 @@
+class Node {
+  constructor(value) {
+    this.value = value;
+    this.children = [];
+  }
+
+  addChild(child) {
+    this.children.push(child);
+  }
+  find(value) {
+    for (let i = 0; i < this.children.length; i++) {
+      if (this.children[i].value == value) {
+        return this.children[i];
+      } else if (this.children[i] == this.children[this.children.length - 1]) {
+        for (let n = 0; n < this.children.length; n++) {
```

Comment:
Why you are iterating over node's children twice?

Reply:
Yes, First to search in children and when it couldnt find it, its start to search level lower from beginning